near2eth/near2eth-block-relay/index.js

```diff
@@ -217,8 +262,35 @@ class Near2EthRelay {
         // console.log({bridgeState, currentBlockHash, lastBlock, replaceDuration}) // DEBUG
         if (bridgeState.currentHeight < lastBlock.inner_lite.height) {
           if (nextValidAt.isZero() || replaceDelay.cmpn(0) <= 0) {
+            // Serialize once here to avoid multiple 'borshify(...)' function calls
+            const blockCouple = {
+              lightClientBlock: lastBlock,
+              borshBlock: borshify(lastBlock)
+            }
+
+            // Describe the selection logic here so as not to complicate the reading of the code
+            if (nextBlockSelection.isEmpty()) {
+              console.log('The selection of the optimal block has begun.')
+              nextBlockSelection.set(blockCouple)
+            } else if (nextBlockSelection.isSuitable(blockCouple)) {
+              nextBlockSelection.set(blockCouple)
+            }
+
+            const selectDelay = selectDuration
+              .add(web3.utils.toBN(nextBlockSelection.startedAt))
+              .sub(web3.utils.toBN(lastBlock.inner_lite.timestamp))
+            if (selectDelay.cmpn(0) > 0) {
+              const selectDelaySeconds = selectDelay.div(web3.utils.toBN(1000_000_000))
+              console.log(`Last light client block: ${lastBlock.inner_lite.height}`)
+              console.log(`Time left to make a decision: ${selectDelaySeconds.toString()} seconds`)
```

Comment:
Let's make sure that each selection iteration takes 1 log at max. Otherwise, the logs will be flooded with info. Also, please add a verbose option to it. So if verbose is off, only the summary (see my other comment) will be printed when the selection is finished.

Make sure to make messages shorter:
- e.g. `"Time left to make a decision: ${selectDelaySeconds.toString()} seconds"` -> `"${selectDelaySeconds.toString()}s left"`

Reply:
As I see the `verbose` option is not implemented at the moment. I will take into account other comments and offer a solution with non-flooding logs in the next commit.